src/jit/codegenxarch.cpp

```diff
@@ -5726,16 +5726,11 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
 #ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
     emitAttr secondRetSize = EA_UNKNOWN;
-    if (varTypeIsStruct(call->gtType))
+    if (call->HasMultiRegRetVal())
     {
-        // Make sure it is a multi-register returned struct,  
-        // otherwise, for a struct passed in a single register   
-        // the call would have a normalized type that is not a struct type.  
-        assert(call->structDesc.passedInRegisters &&
-               (call->structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_RETURN_IN_REGISTERS));
-
-        retSize = emitTypeSize(compiler->getEightByteType(call->structDesc, 0));
-        secondRetSize = emitTypeSize(compiler->getEightByteType(call->structDesc, 1));
+        MultiRegReturnTypeDesc* retTypeDesc = &(call->gtMultiRegReturnTypeDesc);
+        retSize = emitTypeSize(retTypeDesc->GetReturnRegType(1));
+        secondRetSize = emitTypeSize(retTypeDesc->GetReturnRegType(2));
```

Comment:
Previously, at this point it was the case that, if the call node was of a struct type, we were asserting that it must be a multi-reg return - otherwise it would have been transformed. With these changes we seem to have lost that assert.